node/src/database/db_migrations.rs

```diff
@@ -0,0 +1,502 @@
+        mut conn: Box<dyn ConnectionWrapper + 'a>,
+    ) -> Result<(), String> {
+        let updates_to_process = Self::aggregated_checks(mismatched_schema, list_of_updates)?;
+        let mut peekable_list = updates_to_process.iter().peekable();
+        for _ in 0..peekable_list.len() {
+            let (record, versions_in_question) = Self::process_items_from_beneath_dirty_references(
+                peekable_list.next(),
+                peekable_list.peek(),
+            );
+
+            if let Err(e) = record.migrate(&mut conn) {
+                return self.prepare_mailing_of_bad_news(&versions_in_question, e);
+            }
+            self.log_success(&versions_in_question)
+        }
+        Ok(())
+    }
+
+    fn aggregated_checks<'a>(
+        mismatched_schema: &str,
+        list_of_updates: &'a [&'a (dyn MigrateDatabase + 'a)],
+    ) -> Result<Vec<&'a (dyn MigrateDatabase + 'a)>, String> {
+        Self::schema_initial_validation_check(mismatched_schema);
+        let updates_to_process = Self::list_validation_check(list_of_updates)
+            .skip_while(|entry| entry.version_compatibility().ne(mismatched_schema))
+            .map(|e| e.deref())
+            .collect::<Vec<&(dyn MigrateDatabase + 'a)>>();
+        let _ =
+            Self::check_the_number_of_those_remaining(mismatched_schema, updates_to_process.len())?;
+        Ok(updates_to_process)
+    }
+
+    fn process_items_from_beneath_dirty_references<'a>(
+        first: Option<&'a &dyn MigrateDatabase>,
+        second: Option<&&&dyn MigrateDatabase>,
+    ) -> (&'a dyn MigrateDatabase, String) {
+        let first = *first.expect_decent("migration record");
+        let second_by_its_name = if let Some(next) = second {
+            next.version_compatibility()
+        } else {
+            CURRENT_SCHEMA_VERSION
+        };
+        (
+            first,
+            Self::identification_of_versions(first.version_compatibility(), second_by_its_name),
+        )
+    }
+
+    fn check_the_number_of_those_remaining(
+        mismatched_schema: &str,
+        count: usize,
+    ) -> Result<(), String> {
+        match count {
+            0 => Err(format!("Database claims to be more advanced ({}) than the version {} which is the latest released.", mismatched_schema, CURRENT_SCHEMA_VERSION)),
+            _ => Ok(())
+        }
+    }
+
+    fn prepare_mailing_of_bad_news(
+        &self,
+        versions: &str,
+        error: rusqlite::Error,
+    ) -> Result<(), String> {
+        let error_message = format!("Updating database {} failed: {:?}", versions, error);
+        warning!(self.logger, "{}", &error_message);
+        Err(error_message)
+    }
+
+    fn identification_of_versions(first: &str, second: &str) -> String {
+        format!("from version {} to {}", first, second)
+    }
+
+    fn log_success(&self, versions: &str) {
+        info!(self.logger, "Database successfully updated {}", versions)
+    }
+
+    fn list_validation_check<'a>(
```

Comment:
I think your idea of doing this just in the tests, not in production, is a good one for this project.